docs/website/packages/codegen/generate.md

```diff
@@ -0,0 +1,61 @@
+---
+sidebar_position: 1
+title: Generating
+description: codegen's generate function
+---
+
+The generate function loads FHIR structure definition files, and integrates them into typescript code, which is determined by templates that you prepare. These templates are in [EJS format](https://ejs.co/). The files that generate the definitions can be found at `packages/codegen/definitions/fhir`.
+
+## Basic Example
+
+### Prepare your template
+
+Prepare an EJS formatted template, or multiple templates, and store them in a folder called `/templates` in your project, here is a basic example of a template:
+
+```ejs
+export const ResourceNames = [<% for (const structureDef of fhir.resources) { -%>
+  "<%= structureDef.type -%>",
+<% } -%>];
+```
+
+This template will produce an array of resource names.
+
+### Create a codegen script
+
+We must now create a script for codegen in our `package.json``. here is an example:
+
+```json
+...
+  "scripts": {
+    "codegen": "pnpm --dir=\"${PWD}/../codegen\" dev generate --fhir r4b -t \"${PWD}/templates/**/*.ejs\" -b \"${PWD}/templates\" -p 'prettier --write %files%' -o \"${PWD}/your-output-path\"",
+    ...
+  }
+...
+
+```
+
+- Make sure to specify an actual output path for your code in the `-o` command
+- specify the version of Fhir your want to use with `--fhir`
+
+### Run the Script
+
+run your script within package that your created it with `pnpm codegen`
+
+### Test Generated code In Your Project
+
+Your code with be added to your output destination in the file format `**.codegen.ts`
+
+the example template above will produce something like this:
+
+```ts
+export const ResourceNames = [
+  "Account",
+  "ActivityDefinition",
+  "AdministrableProductDefinition",
+  "AdverseEvent",
+  "AllergyIntolerance",
+  "Appointment",
+  "AppointmentResponse",
+  "AuditEvent",
+...
+```
```

Comment:
+ add:
 - the fact that the final file name is the name of the template minus the extension (e.g. `resource-names.ts.ejs` -> `resource-names.ts`)
 - `fhir` is the variable injected into the context that has the FHIR structure definitions loaded - here is the structure of the data model: https://github.com/bonfhir/bonfhir/blob/main/packages/codegen/src/fhir/fhir-definitions.ts
 - you can specify multiple fhir version at once: `r5,r4b` - in that case, the output path can also be templated - e.g. `-o "./src/<%= fhir.release %>"`